server/odc-service/src/main/java/com/oceanbase/odc/service/db/schema/DBSchemaIndexService.java

```diff
@@ -201,6 +203,12 @@ public Boolean syncDatabaseObjects(@NonNull @Valid SyncDBObjectReq req) {
         this.databaseService.refreshExpiredPendingDBObjectStatus();
         Set<Database> databases = new HashSet<>();
         if (req.getResourceType() == ResourceType.ODC_CONNECTION) {
+            try {
+                databaseService.internalSyncDataSourceSchemas(req.getResourceId());
+            } catch (InterruptedException ex) {
```

Comment:
catch exception instead

Reply:
It is recommended to return the error directly if not getting the lock, because if not getting the lock, it may not be the latest database collection in odc, so it is meaningless to synchronize the database object like table ,view etc later.
What's more, even if successful, the list of databases is still not up to date.